Mathlib/Topology/CWComplex/Basic.lean

```diff
@@ -0,0 +1,682 @@
+/-
+Copyright (c) 2024 Floris van Doorn and Hannah Scholz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn, Hannah Scholz
+-/
+
+import Mathlib.Analysis.InnerProductSpace.Basic
+import Mathlib.Data.Real.StarOrdered
+import Mathlib.Order.CompletePartialOrder
+
+/-!
+# CW-complexes
+
+This file defines (relative) CW-complexes and proofs basic properties about them.
+
+A CW-complex is a topological space that is made by gluing closed disks of different dimensions
+together.
+
+## Main definitions
+* `RelCWComplex C D` : The class of CW-structures on a subspace `C` realtive to `D` of a topological
+  space `X`.
+* `ClasCWComplex C` : An abbreviation for `RelCWComplex C ∅`. The class of CW-structures on a
+  subspace `C` of the topological space `X`.
+* `openCell n i` : An open cell of dimension `n`.
+* `closedCell n i` : A closed cell of dimension `n`.
+* `cellFrontier n i` : The edge of a cell of dimension `n`.
+* `level C n` : The `n`-th level of the (relative) CW-complex `C`.
+
+## Main statements
+* `iUnion_openCell_eq_level` : The levels can also be seen as a union of open cells.
+* `cellFrontier_subset_finite_openCell` : The edge of a cell is contained in a finite union of
+  open cells of a lower dimension.
+
+## Implementation notes
+* This definition uses is the historical definition rather than the modern one to avoid working
```

Comment:
How about rewriting the first two bullet points, a bit like the following? (Is what I am saying correct? I didn't read the implementation yet.)

We use the historical definition of CW complexes, due to Whitehead: a CW complex is a collection of cells with attaching maps --- all cells are subspaces of one ambient topological space. This way, we avoid having to work with a log of different topological spaces. On the other hand, it requires the subspaces to be closed.
If that is not the case, you need to consider that subspace as a subspace of itself.